lld/wasm/SymbolTable.cpp

```diff
@@ -526,6 +535,7 @@ Symbol *SymbolTable::addUndefinedFunction(StringRef name,
     printTraceSymbolUndefined(name, file);
```

Comment:
We run `printTraceSymbolUndefined` here and possibly again when `wasInserted` is true below. Does this mean `s->traced` and `wasInserted` are never true at the same time?
Would it be clear if we check two conditions at a single place to run this?

Reply:
No, `s->traced` depends only on the command line flags and will be true if and only if `--trace-symbol=foo` is passed for a given symbol `foo`.   It can be true or false independently of everything else going on here.

I'm not sure what you mean by "and then again".  I only see a single call to `printTraceSymbolUndefined` in this function.

Reply:
This function contains a call to `traceImport` which calls `printTraceSymbolUndefined`. So I guess if `s->traced` and `wasInserted` are both true it can be called twice?

Reply:
Ah yes in that case you potentially see a trace entry for the "name" and the "import name".  Not that those are normally different but if you happened to trace both then I would expect two trace entires in the log yes.